compiler/qsc_rir/src/passes/reindex_qubits.rs

```diff
@@ -187,27 +191,37 @@ impl ReindexQubitPass {
                         })
                         .collect::<Vec<_>>();
 
-                    if call_id == self.m_id {
-                        // Since the call was to mz, the new qubit replacing this one must be conditionally flipped.
-                        // Achieve this by adding a CNOT gate before the mz call.
-                        self.used_cx = true;
-                        block.0.push(Instruction::Call(
-                            self.cx_id,
-                            vec![
-                                new_args[0],
-                                Operand::Literal(Literal::Qubit(qubit_map.next_qubit_id)),
-                            ],
-                            None,
-                        ));
-                        self.highest_used_id = self.highest_used_id.max(qubit_map.next_qubit_id);
+                    if *call_id == self.m_id {
+                        if qubit_used_in_instrs(ids_used[0], &instrs[i + 1..]) {
```

Comment:
is the indexing of `[i + 1..]` safe? Will this crash if we're on the last instr?

Reply:
Switched it to using iterators.